src/plugins/clips-executive/clips/domain.clp

```diff
@@ -596,6 +597,11 @@
         (goal-id ?g) (plan-id ?p)
         (part-of ?pn) (grounded TRUE)
         (grounded-with ?action-id) (is-satisfied FALSE)))
+
+  ; ensure that we have grounded all related atomic preconditions before
+  (forall (domain-atomic-precondition (operator ?op) (part-of ?pn) (name ?apname) (grounded FALSE))
+    (domain-atomic-precondition (operator ?op) (part-of ?pn) (name ?apname) (grounded TRUE))
```

Comment:
We should make sure that the grounding is for the right action:
```suggestion
    (domain-atomic-precondition (operator ?op) (part-of ?pn) (name ?apname) (grounded TRUE) (grounded-with ?action-id))
```

Reply:
This is a good point, I started wondering this in the morning and was about to write a test case with multiple calls to the same operator in a plan. Applying this fix resolves exactly this.